web/source/osk/oskManager.ts

```diff
@@ -820,8 +807,17 @@ namespace com.keyman.osk {
       // TODO:  Move this into the VisualKeyboard class!
       // The following code will always be executed except for externally created OSK such as EuroLatin
       if(this.vkbd && this.vkbd.ddOSK) {
+        // Always adjust screen height if iPhone or iPod, to take account of viewport changes
+        // Do NOT condition upon form-factor; this line prevents a bug with displaying
+        // the predictive-text banner on the initial keyboard load.  (Issue #2907)
+        if(device.touchable && device.OS == 'iOS') {
+          this.vkbd.adjustHeights(this.getKeyboardHeight());
```

Comment:
This particular call (`adjustHeights`) is no longer included in `VisualKeyboard`'s `show` / `updateState` method.